### ISSUE TITLE
When deleting app from Android device, unable to successfully reinstall using flutter run

### ISSUE THREAD

**Issue by Vortexmind** [flutter/flutter#28780]
When I create the sample app from the [tutorial](https://flutter.dev/docs/get-started/test-drive), and either leave it as-is, or modify it with valid code, `flutter run` works fine with my Android device. However, if I uninstall the app from the device, and then attempt to run `flutter run` again, I get an error despite the output of the command telling me that the app was installed successfully on the device 
```
flutter run
Launching lib/main.dart on [Device] in debug mode...
Initializing gradle...                                              1.4s
Resolving dependencies...                                           3.8s
Running Gradle task 'assembleDebug'...
Running Gradle task 'assembleDebug'... Done                         3.0s
Built build\app\outputs\apk\debug\app-debug.apk.
Installing build\app\outputs\apk\app.apk...                         6.0s

(hangs)
```

Looking at a more detailed output (I removed some details about my paths / app name)
```
flutter run -v
[...]
[        ] executing: C:\[...]\sdk\platform-tools\adb -s [...]install -t -r C:\[...]\build\app\outputs\apk\app.apk
[+7554 ms] Success
[        ] Installing build\app\outputs\apk\app.apk... (completed in 7.6s)
[   +1 ms] executing: C:\[...]\sdk\platform-tools\adb -s [...]shell echo -n 8b761cbba713cd85353fada42c430fcf4f1e6acb > /data/local/tmp/sky.com.example.[...]
[  +60 ms] [Device] startApp
[   +1 ms] executing: C:\[...]\sdk\platform-tools\adb -s [...]shell am start -a android.intent.action.RUN -f 0x20000000 --ez enable-background-compilation true --ez
enable-dart-profiling true --ez enable-checked-mode true --ez verify-entry-points true com.example.[...]/com.example[...].MainActivity
[ +102 ms] Starting: Intent { act=android.intent.action.RUN flg=0x20000000 cmp=com.example.[...]/.MainActivity (has extras) }

                    Error type 3
                    Error: Activity class {com.example.[...]/com.example.[...].MainActivity} does not exist.
[        ] Waiting for observatory port to be available...
```
In fact - if I look on the device, the app is not listed.

I can resolve the issue using `adb`

```
> adb install [...]\build\app\outputs\apk\app.apk
adb: failed to install [...]\build\app\outputs\apk\app.apk: Failure [INSTALL_FAILED_ALREADY_EXISTS: Attempt to re-install com.example.[...] without first uninstalling.]
> adb uninstall com.example.[...]
Success
> adb install [...]\build\app\outputs\apk\app.apk
Success
```
At this point - the app appears on the device

If I run `flutter run` after this (or launch the debug mode in Visual Studio code) then I can continue working. This was reproduced both on Windows and on Linux (Ubuntu) development environment 

```
flutter analyze
Analyzing [app]...
No issues found! (ran in 2.4s)


flutter doctor -v
[√] Flutter (Channel stable, v1.2.1, on Microsoft Windows [Version 10.0.17134.590], locale en-GB)
    • Flutter version 1.2.1 at C:\flutter
    • Framework revision 8661d8aecd (2 weeks ago), 2019-02-14 19:19:53 -0800
    • Engine revision 3757390fa4
    • Dart version 2.1.2 (build 2.1.2-dev.0.0 0a7dcf17eb)

[√] Android toolchain - develop for Android devices (Android SDK version 28.0.3)
    • Android SDK at C:\[...]\AppData\Local\Android\sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-28, build-tools 28.0.3
    • Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1248-b01)
    • All Android licenses accepted.

[!] Android Studio (version 3.3)
    • Android Studio at C:\Program Files\Android\Android Studio
    X Flutter plugin not installed; this adds Flutter specific functionality.
    X Dart plugin not installed; this adds Dart specific functionality.
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1248-b01)

[√] VS Code (version 1.31.1)
    • VS Code at C:\[...]\Local\Programs\Microsoft VS Code
    • Flutter extension version 2.24.0

[√] Connected device (1 available)
    • [...] • [...] • android-arm • Android 8.1.0 (API 27)

! Doctor found issues in 1 category.
```

**Comment by jonahwilliams** [comment]
I've tried a few combinations of installing/uninstalling/unplugging/adb uninstalling and I'm unable to reproduce this on 1.2.1 or master. Was this a real device (particular mode?) or an emulator?

**Comment by s3cpat** [comment]
I've run into the same issue when using a real device (Google Pixel 2)

**Comment by Vortexmind** [comment]
@jonahwilliams - sorry for the delay. The issue occurred on a Motorola Moto G5 Plus . I have just reproduced it after updating flutter to latest, here's the output of `flutter doctor`

```
>flutter doctor -v
[√] Flutter (Channel stable, v1.5.4-hotfix.2, on Microsoft Windows [Version 10.0.17134.706], locale en-GB)
    • Flutter version 1.5.4-hotfix.2 at C:\flutter
    • Framework revision 7a4c33425d (10 days ago), 2019-04-29 11:05:24 -0700
    • Engine revision 52c7a1e849
    • Dart version 2.3.0 (build 2.3.0-dev.0.5 a1668566e5)


[√] Android toolchain - develop for Android devices (Android SDK version 28.0.3)
    • Android SDK at C:\[...]\AppData\Local\Android\sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-28, build-tools 28.0.3
    • Java binary at: C:\[...]\Android Studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1343-b01)
    • All Android licenses accepted.

[√] Android Studio (version 3.4)
    • Android Studio at C:\[...]\Android Studio
    • Flutter plugin version 35.0.1
    • Dart plugin version 183.6270
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1343-b01)

[√] VS Code (version 1.31.1)
    • VS Code at C:\[...]\Microsoft VS Code
    • Flutter extension version 2.24.0

[√] Connected device (1 available)
    • Moto G 5 Plus •  [...] • android-arm • Android 8.1.0 (API 27)

• No issues found!
```

**Comment by pram507** [comment]
i have same issue like this. after build apk and install -> uninstall -> install. apk will cant installed.

**Comment by vikasgola** [comment]
I am also facing the same issue.

**Comment by JocelynDavid** [comment]
Same here . 
I installed first version without problem (build an apk and share it on slack to download on other devices). I uninstall it, and share a new one, impossible to start it (app not started)

**Comment by vikasgola** [comment]
@JocelynDavid have you signed the apk?

Unsigned apks can't be installed on Android without adb.

**Comment by voznyy** [comment]
I've also faced the same issue. The only way for me to solve this was to uninstall the app from ADB:
`adb uninstall com.package.name`

**Comment by nanodesu88** [comment]
@voznyy it works!

**Comment by marcellocbastos** [comment]
Appear the same error to me... But I uninstalled the App dragging the App Icon to option "Uninstall"... So I was trying run again and it was building and not running in Mobile. But the answer  from [Ritam Chakraborty](https://stackoverflow.com/users/10644028/ritam-chakraborty) in [StackOverflow](https://stackoverflow.com/questions/51809421/flutter-is-not-able-to-install-the-apk-into-the-real-device-suddenly/62650276#62650276) did I look for the App in "Applications" and I found there! So, accessing App, clicking in "Uninstall" and debugging or running again on VS Code it worked! Maybe can help someone...
